src/modbus.c

```diff
@@ -983,11 +1003,26 @@ static int mb_config_add_host(oconfig_item_t *ci) /* {{{ */
         status = -1;
     } else if (strcasecmp("Device", child->key) == 0) {
       status = cf_util_get_string_buffer(child, host->node, sizeof(host->node));
-      if (status == 0)
+      if (status == 0) {
         host->conntype = MBCONN_RTU;
+        host->uarttype = UARTTYPE_RS232;
+      }
     } else if (strcasecmp("Baudrate", child->key) == 0)
       status = cf_util_get_int(child, &host->baudrate);
-    else if (strcasecmp("Interval", child->key) == 0)
+    else if (strcasecmp("UARTType", child->key) == 0) {
```

Comment:
This configuration directive should be processed with respect to `LEGACY_LIBMODBUS` define.

```
#if !LEGACY_LIBMODBUS && LIBMODBUS_VERSION_CHECK(2,9,4)
      [... your code ...]
#else
      ERROR("Modbus plugin: Option `UARTType' not supported. Please "
            "upgrade libmodbus to at least 2.9.4");
      return -1;
#endif
```

This also should check `#if defined(linux)` too.